src/wasm/wasm-binary.cpp

```diff
@@ -1511,23 +1524,27 @@ void WasmBinaryBuilder::readTableElements() {
 
 void WasmBinaryBuilder::readNames() {
   if (debug) std::cerr << "== readNames" << std::endl;
+  auto name_type = getU32LEB();
+  auto subsection_size = getU32LEB();
+  WASM_UNUSED(subsection_size);
```

Comment:
I see, thanks. I'd be curious to see how the binaryen optimizer works on your output. It's a goal for it to be as general purpose as possible, so testing on more compilers can help.